Distribution/Server/Features/PackageInfoJSON/State.hs

```diff
@@ -0,0 +1,251 @@
+{-# LANGUAGE DeriveDataTypeable  #-}
+{-# LANGUAGE DeriveGeneric       #-}
+{-# LANGUAGE LambdaCase          #-}
+{-# LANGUAGE RecordWildCards     #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TemplateHaskell     #-}
+{-# LANGUAGE TypeFamilies        #-}
+
+module Distribution.Server.Features.PackageInfoJSON.State where
+
+import           Control.Arrow        (first, second)
+import           Control.Applicative  ((<|>))
+import           Control.Monad.Reader (ask, asks)
+import qualified Control.Monad.State  as State
+import qualified Data.Aeson           as Aeson
+import           Data.Aeson           ((.=), (.:))
+import           Data.Acid            (Query, Update, makeAcidic)
+import qualified Data.HashMap.Strict  as HashMap
+import qualified Data.Map             as Map
+import           Data.Monoid          (Sum(..))
+import qualified Data.Text            as T
+import qualified Data.Text.Encoding   as T
+import           Data.SafeCopy        (SafeCopy(..), base, contain,
+                                       deriveSafeCopy)
+import           Data.Serialize       (Get, get, getListOf, getTwoOf, put,
+                                       putListOf, putTwoOf)
+import           Data.Typeable        (Typeable)
+import           Data.Word            (Word8)
+import           Distribution.License (licenseToSPDX)
+import           Distribution.Text    (display, simpleParse)
+import           GHC.Generics         (Generic)
+
+import           Distribution.SPDX.License (License)
+import           Distribution.Package      (PackageIdentifier, PackageName)
+import           Distribution.Version      (Version, mkVersion, versionNumbers)
+import qualified Distribution.Pretty       as Pretty
+import qualified Distribution.Parsec.Class as Parsec
+
+import qualified Distribution.Server.Features.PreferredVersions as Preferred
+import           Distribution.Server.Framework.MemSize          (MemSize,
+                                                                 memSize,
+                                                                 memSize7)
+
+
+-- | Basic information about a package. These values are
+--   used in the `/package/:packagename` JSON endpoint
+data PackageBasicDescription = PackageBasicDescription
+  { pbd_license           :: License
```

Comment:
Can we make these fields strict? We put that into a map in memory, it would be a shame to introduce a leak.